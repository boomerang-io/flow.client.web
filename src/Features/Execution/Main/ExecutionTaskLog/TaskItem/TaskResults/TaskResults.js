import React from "react";
import PropTypes from "prop-types";
import ReactJson from "react-json-view";
import { Button, ComposedModal, ModalForm } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody, Tabs, Tab } from "@boomerang-io/carbon-addons-boomerang-react";
import ResultsTable from "./ResultsTable";
import styles from "./taskResults.module.scss";

function TaskResults({ flowTaskName, flowTaskResults }) {

  return (
    <ComposedModal
      composedModalProps={{ containerClassName: styles.container, shouldCloseOnOverlayClick: true }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved",
      }}
      modalHeaderProps={{
        title: "Task Results",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }) => (
        <Button kind="ghost" size="small" onClick={openModal}>
          View Results
        </Button>
      )}
    >
      {() => (
        <ModalForm>
          <ModalBody>
            <Tabs>
              <Tab label="Table">
                <ResultsTable data={flowTaskResults} />
              </Tab>
              <Tab label="JSON">
                <div className={styles.propertiesJson}>
                  <ReactJson
                    name={false}
                    src={flowTaskResults}
                    displayDataTypes={false}
                    enableDelete={false}
                    displayObjectSize={false}
                    enableEdit={false}
                    enableAdd={false}
                  />
                </div>
              </Tab>
            </Tabs>
          </ModalBody>
        </ModalForm>
      )}
    </ComposedModal>
  );
}

TaskResults.propTypes = {
  flowTaskName: PropTypes.string.isRequired,
  flowTaskResults: PropTypes.object.isRequired,
};

export default TaskResults;
