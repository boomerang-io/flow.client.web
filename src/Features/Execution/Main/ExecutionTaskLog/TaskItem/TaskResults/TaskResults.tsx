import React from "react";
import { Button, ComposedModal, ModalForm } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody } from "@boomerang-io/carbon-addons-boomerang-react";
import ResultsTable from "./ResultsTable";
import styles from "./taskResults.module.scss";

type Props = {
  flowTaskName: string;
  flowTaskResults: {
    name: string;
    description: string;
    value: string;
  }[];
};

function TaskResults({ flowTaskName, flowTaskResults }: Props) {
  return (
    <ComposedModal
      composedModalProps={{ containerClassName: styles.container, shouldCloseOnOverlayClick: true }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved",
      }}
      modalHeaderProps={{
        title: "Task Results",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }: { openModal: () => void }) => (
        <Button kind="ghost" size="small" onClick={openModal}>
          View Results
        </Button>
      )}
    >
      {() => (
        <ModalForm>
          <ModalBody>
            <ResultsTable data={flowTaskResults} />
          </ModalBody>
        </ModalForm>
      )}
    </ComposedModal>
  );
}

export default TaskResults;
