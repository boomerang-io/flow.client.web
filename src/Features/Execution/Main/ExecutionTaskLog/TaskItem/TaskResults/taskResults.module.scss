@import "Styles/mixins";

.container {
  max-width: 50rem;

  :global .bx--modal-content {
    padding-right: 1rem;

    .bx--tab-content {
      padding: 0;
    }
  }
}


.propertiesJson {
  background-color: var(--cds-ui-01);
  height: 23rem;
  overflow-y: auto;
  margin-top: 1rem;

  :global .object-key {
    span {
      color: var(--cds-interactive-01);
    }
  }
}
