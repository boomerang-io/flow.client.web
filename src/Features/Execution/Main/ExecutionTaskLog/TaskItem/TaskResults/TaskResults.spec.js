import React from "react";
import TaskResults from "./index";
import { queryCaches } from "react-query";

afterEach(() => {
  queryCaches.forEach((queryCache) => queryCache.clear());
});

const props = {
  flowTaskName: "Send Slack Message",
  flowTaskResults: [
    {
      "name": "test",
      "description": "Test description",
      "value": "Testing value"
    }
  ],
};

describe("TaskResults --- Snapshot", () => {
  it("Capturing Snapshot of TaskResults", () => {
    const { baseElement } = rtlRender(<TaskResults {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
