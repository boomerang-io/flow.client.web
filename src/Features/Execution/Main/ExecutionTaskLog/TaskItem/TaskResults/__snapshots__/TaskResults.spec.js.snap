// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskResults --- Snapshot Capturing Snapshot of TaskResults 1`] = `
<body
  id="app"
>
  <div>
    <button
      class="bx--btn bx--btn--sm bx--btn--ghost"
      tabindex="0"
      type="button"
    >
      View Results
    </button>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
